nodes/vector/vector_polar_out.py

```diff
@@ -54,9 +54,9 @@ class VectorPolarOutNode(bpy.types.Node, SverchCustomTreeNode):
     ]
 
     def coordinate_changed(self, context):
-        self.outputs["z"].hide = self.coordinates == "theta"
-        self.outputs["theta"].hide = self.coordinates == "z"
-        updateNode(self, context)
+        if self.coordinates != self.outputs[2].name:
```

Comment:
I do not like magic numbers :) And, the idea of relying on node's property being equal to output name is not great imho. Just because it is not explicit.
I'd prefer more explicit (even if more lines of code):
````
if self.coordinates == "z":
...
elif self.coordinates == "theta":
...
else:
    # Just in case we will add something later and forget to add a check here
    raise Exception("Unsupported coordinates type: {}".format(self.coordinates))
````

Reply:
Huh, I just discovered I used the same idea in another part of this code :) Well, I still do not like it.
As an option, we could do
````
def get_third_coordinate_name(self):
    # Here we are relying on self.corodinates being equal to name of output, but now it is explicit
    return self.coordinates
````
and then use this method , for example, in line 110 and similar places:

````
 if self.outputs[self.get_third_coordinate_name()].is_linked:
````